docs/root/configuration/http_filters/jwt_authn_filter.rst

```diff
@@ -0,0 +1,214 @@
+
+JwtProvider config example one
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: yaml
+
+  providers:
+    provider_name1:
+      issuer: https://example.com
+      audiences:
+      - bookstore_android.apps.googleusercontent.com
+      - bookstore_web.apps.googleusercontent.com
+      remote_jwks:
+        http_uri:
+          uri: https://example.com/jwks.json
+          cluster: example_jwks_cluster
+        cache_duration:
+          seconds: 300
+
+Above example specifies:
+
+* *provider_name* is **provider_name1**
+* *issuer* is **https://example.com**
+* *audiences*: **bookstore_android.apps.googleusercontent.com** and **bookstore_web.apps.googleusercontent.com**
+* *remote_jwks*: use URL **https://example.com/jwks.json** from cluster **example_jwks_cluster**, its cache duration is 300 seconds.
+* token will be extracted from the default extract locations since *from_headers* and *from_params* are empty.
+* token will not be forwarded to upstream since *forward* is not set, default is **false**
+* JWT payload will not be added to the request header since *forward_payload_header* is not set.
+
+Following cluster **example_jwks_cluster** is needed to fetch JWKS.
+
+.. code-block:: yaml
+
+  cluster:
+    name: example_jwks_cluster
+    type: STRICT_DNS
+    hosts:
+      socket_address:
+        address: example.com
+        portValue: 80
+
+
+JwtProvider config example two
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Another config example using inline JWKS:
+
+.. code-block:: yaml
+
+  providers:
+    provider_name2:
+      issuer: https://example2.com
+      local_jwks:
+        inline_string: "PUBLIC-KEY"
+      from_headers:
+      - name: jwt-assertion
+      forward: true
+      forward_payload_header: x-jwt-payload
+
+Above example specifies:
+
+* *provider_name* is **provider_name2**
+* *issuer* is **https://example2.com**
+* *audiences*: not specified, JWT *aud* field will not be checked.
+* *local_jwks*: JWKS is embeded in the inline string.
+* *from_headers*: token will be extracted from HTTP headers as::
+
+     jwt-assertion: <JWT>.
+
+* *forward*: token will forwarded to upstream
+* JWT payload will be added to the request header as following format::
+
+    x-jwt-payload: base64_encoded(jwt_payload_in_JSON)
+
+RequirementRule
+~~~~~~~~~~~~~~~
+
+:ref:`RequirementRule <envoy_api_msg_config.filter.http.jwt_authn.v2alpha.RequirementRule>` has two fields: *match* and *requires*. The field *match* specifies how a request can be matched; e.g. by HTTP headers, or by query parameters, or by path prefixes. The field *requires* specifies the JWT requirement, e.g. which provider is required. Multiple providers may be required in such complex forms as "require_all" or "require_any".
```

Comment:
Something like this seems to make sense to do inline in the API docs rather than independent RST.

Reply:
Removed most of them